AssettoServer/Commands/Modules/AdminModule.cs

```diff
@@ -108,11 +110,29 @@ public void SetWeather(int weatherId)
         }
     }
 
+    [Command("cspweather")]
+    public void CspWeather()
+    {
+        List<WeatherFxType> weathers = Enum.GetValues<WeatherFxType>().ToList();
+        Context.Client.SendPacket(new ChatMessage { SessionId = 255, Message = "Available weathers:" });
+        foreach (WeatherFxType weather in weathers)
+        {
+            Context.Client.SendPacket(new ChatMessage { SessionId = 255, Message = $" - {weather}" });
+        }
+    }
+
     [Command("setcspweather")]
-    public void SetCspWeather(int upcoming, int duration)
+    public void SetCspWeather(string upcomingWeatherName, int duration)
```

Comment:
you can change the parameter type from `string` to `WeatherFxType` and Qmmands will handle the enum parsing for you

Reply:
hm you're right, the custom message with a reference to the /cspweather command is much better here. Feel free to re-add that part